packages/tangle/parser.go

```diff
@@ -246,6 +256,8 @@ type MessageFilter interface {
 	OnAccept(callback func(msg *Message, peer *peer.Peer))
 	// OnAccept registers the given callback as the rejection function of the filter.
 	OnReject(callback func(msg *Message, err error, peer *peer.Peer))
+	// Close closes the filter.
+	Close()
```

Comment:
You can simply embed `io.Closer`. Probably better as it also defines Close() to potentially return an error.